### ISSUE TITLE
Haproxy Ingress Controller v1.7.x crashes when using backend crts

### ISSUE THREAD

**Issue by dani-CO-CN** [haproxytech/kubernetes-ingress#402]
I upgraded the haproxy ingress controller using the helm chart from version 1.15.1 to the helm chart version 1.17.11. When an ingress resources defines a  server-crt the controller crashes, once the annotation is removed the controller starts up. I use EKS v1.21.2

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    haproxy.org/server-crt: argocd/argocd-secret
```
haproxy log:
```
2021/12/10 11:28:02 ERROR   controller.go:190 unable to Sync HAProxy configuration !!
2021/12/10 11:28:02 ERROR   controller.go:191 14: err transactionId=b26164c2-cb4b-4808-8265-94d0a3392c9b 
```
The ingress controller then stays in a crash loop, i experimented a bit with the versions of the ingress controller. This error was introduced between helm chart version 1.16.3 and 1.17.0. With 1.16.3 everything starts without problems and with the haproxy ingress version 1.7.0 it no longer works. Below are my helm values.
```yaml
defaultBackend:
  false
controller:
  kind: DeamonSet
  ingressClass: somename
  defaultTLSSecret:
    enabled: false
  daemonset:
    useHostPort: true
  publishService:
    enabled: false
  config: 
    ssl-redirect: "true"
    syslog-server: address:stdout, format: raw, facility:daemon
  service:
    enabled: false
    tcpPorts:
    - name: placeholder
      port: placeholder
      nodePort: placeholder
  extraArgs:
  - '--default-ssl-certificate=namespace/some-certificate'
  - '--configmap-tcp-services=namespace/a-configmap'
```

**Comment by Mo3m3n** [comment]
Thanks @dani-CO-CN  for reporting this issue.
It seems the problem is related to the core haproxy software.
I have opened a ticket in the corresponding repository https://github.com/haproxy/haproxy/issues/1488

**Comment by Mo3m3n** [comment]
Should be fixed in `v1.7.4` built against the latest haproxy 2.4 which [fixes](https://github.com/haproxy/haproxy/commit/2c776f1c30c85be11c9ba8ca8d9a7d62690d1a32) this issue 
Don't hesitate to reopen this issue if it is not the case.
